apps/src/templates/rubrics/productTourHelpers.js

```diff
@@ -1,41 +1,62 @@
+import React from 'react';
+
 import i18n from '@cdo/locale';
 import evidenceDemo from '@cdo/static/ai-evidence-demo.gif';
 
+// intro.js-react allows a string or a react component for the intro prop.
+// Providing a string that was written by a translator is risky, because it
+// could contain malicious HTML. This helper method wraps the string in a react
+// fragment, which will take care of sanitizing the string.
+const sanitize = unsafe => <>{unsafe}</>;
+
 export const INITIAL_STEP = 0;
 export const STEPS = [
   {
     element: '#ui-floatingActionButton',
     title: i18n.rubricTourStepOneTitle(),
-    intro: i18n.rubricTourStepOneText(),
+    intro: sanitize(i18n.rubricTourStepOneText()),
   },
   {
     element: '#tour-ai-assessment',
     title: i18n.rubricTourStepTwoTitle(),
-    intro: i18n.rubricTourStepTwoText(),
+    intro: sanitize(i18n.rubricTourStepTwoText()),
   },
   {
     element: '#tour-ai-evidence',
     title: i18n.rubricTourStepThreeTitle(),
     position: 'top',
-    intro: `<p>${i18n.rubricTourStepThreeText()}</p><img src=${evidenceDemo}>`,
+    intro: (
+      <>
+        <p>{i18n.rubricTourStepThreeText()}</p>
+        <img src={evidenceDemo} alt={i18n.rubricTourStepThreeAltText()} />
+      </>
+    ),
   },
   {
     element: '#tour-ai-confidence',
     title: i18n.rubricTourStepFourTitle(),
-    intro: i18n.rubricTourStepFourText(),
+    intro: sanitize(i18n.rubricTourStepFourText()),
   },
   {
     element: '#tour-evidence-levels',
     title: i18n.rubricTourStepFiveTitle(),
-    intro: i18n.rubricTourStepFiveText(),
+    intro: sanitize(i18n.rubricTourStepFiveText()),
   },
   {
     element: '#tour-ai-assessment-feedback',
     title: i18n.rubricTourStepSixTitle(),
-    intro: i18n.rubricTourStepSixText(),
+    intro: sanitize(i18n.rubricTourStepSixText()),
   },
 ];
 
+STEPS.forEach((step, index) => {
+  if (typeof step.intro === 'string') {
+    throw new Error(
```

Comment:
nit: I suspect this would cause either the tours/the whole page to crash? I wonder if a we did have a malicious string we could possibly blank it out instead of throwing since the i18n could result in a denial of service if this happened. Looks good though!

Reply:
Yes, this should crash the main thread. I'm fairly sure that this would happen regardless of the localized string and instead just ensures that you have wrapped the string in the `sanitize` call. Seems reasonable, but could also be something you write a test for instead, considering we export `STEPS`.

Reply:
Good points, I've moved this to a unit test. if you forget to sanitize like this
```diff
diff --git a/apps/src/templates/rubrics/productTourHelpers.js b/apps/src/templates/rubrics/productTourHelpers.js
index 5adc88ca99e..cc2b7780ce5 100644
--- a/apps/src/templates/rubrics/productTourHelpers.js
+++ b/apps/src/templates/rubrics/productTourHelpers.js
@@ -14,7 +14,7 @@ export const STEPS = [
   {
     element: '#ui-floatingActionButton',
     title: i18n.rubricTourStepOneTitle(),
-    intro: sanitize(i18n.rubricTourStepOneText()),
+    intro: i18n.rubricTourStepOneText(),
   },
```
the test will catch it like this
![Screenshot 2024-07-25 at 8 54 34 PM](https://github.com/user-attachments/assets/883d5864-cbea-45ef-bef0-d1927f2b1442)